src/borg/testsuite/archiver.py

```diff
@@ -1298,9 +1298,12 @@ def test_list_size(self):
         self.cmd('init', self.repository_location)
         test_archive = self.repository_location + '::test'
         self.cmd('create', '-C', 'lz4', test_archive, 'input')
-        output = self.cmd('list', '--format', '{size} {csize} {path}{NL}', test_archive)
-        size, csize, path = output.split("\n")[1].split(" ")
+        output = self.cmd('list', '--format', '{size} {csize} {dsize} {dcsize} {path}{NL}', test_archive)
+        size, csize, dsize, dcsize, path = output.split("\n")[1].split(" ")
         assert int(csize) < int(size)
+        assert int(dcsize) < int(dsize)
+        assert int(dsize) <= int(size)
+        assert int(dcsize) <= int(csize)
```

Comment:
just imagine an archive only has 1 file, a big file with multiple chunks of all zeros.
your code would say dsize == 0 (because there is no chunk with refcount 1).
but in fact dsize should be the size of 1 "all zeros" chunk.

(this is assuming that file size is a precise multiple of the zero chunk size. if that is not the case, the smaller all-zeros chunk at the end would have refcount == 1).